libs/application/templates/children-residence-change/src/dataProviders/MockNationalRegistryProvider.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Application,
+  BasicDataProvider,
+  FailedDataProviderResult,
+  SuccessfulDataProviderResult,
+} from '@island.is/application/core'
+
+import { PersonResidenceChange } from '@island.is/application/templates/children-residence-change'
+import { NationalRegistry, DataProviderTypes, CRCApplication } from '../types'
+
+export class MockNationalRegistryProvider extends BasicDataProvider {
+  readonly type = DataProviderTypes.MOCK_NationalRegistry
+
+  async provide(application: Application): Promise<NationalRegistry> {
+    const query = `
+      query NationalRegistryUserQuery {
+        nationalRegistryUser {
+          nationalId
+          fullName
+          address {
+            code
+            postalCode
+            city
+            streetAddress
+            lastUpdated
+          }
+        }
+      }
+    `
+    const crcApplication = (application as unknown) as CRCApplication
+    const {
+      answers: {
+        mockData: { parents, children },
+      },
+    } = crcApplication
+    if (!children) {
+      throw new Error('Engin börn fundust í þinni forsjá')
+    }
+    const childrenArray: NationalRegistry['children'] = []
+    children?.map((child) => {
+      childrenArray.push({
+        ...child,
+        livesWithApplicant: child?.livesWithApplicant?.includes('yes') || false,
+        otherParent: parents[child.otherParent],
+      })
+    })
+    return this.useGraphqlGateway(query)
+      .then(async (res: Response) => {
+        const response = await res.json()
+        if (response.errors) {
+          throw new Error('Error from NationalRegistry')
```

Comment:
Don't we always end up here since the endpoint is not ready? Or am I missing something? 😅

Reply:
No, we are using the old endpoint from national registry to populate the applicant info.